src/Init/Data/List/MinMax.lean

```diff
@@ -38,6 +39,18 @@ theorem isSome_min?_of_mem {l : List α} [Min α] {a : α} (h : a ∈ l) :
     l.min?.isSome := by
   cases l <;> simp_all [List.min?_cons']
 
+theorem min?_eq_head? {α : Type u} [Min α] {l : List α}
+    (h : l.Pairwise (fun a b => min a b = a)) : l.min? = l.head? := by
```

Comment:
I think it is safe to assume that we will want some vast generalization of this at some point (something like `List.IsPartitionedAt`), but this will do for now.